var/spack/repos/builtin/packages/xz/package.py

```diff
@@ -63,7 +70,63 @@ def determine_version(cls, exe):
         match = re.search(r"xz \(XZ Utils\) (\S+)", output)
         return match.group(1) if match else None
 
+
+class AutotoolsBuilder(AutotoolsBuilder):
+    def configure_args(self):
+        return self.enable_or_disable("libs")
+
     @run_after("install")
     def darwin_fix(self):
         if self.spec.satisfies("platform=darwin"):
             fix_darwin_install_name(self.prefix.lib)
+
+
+class MSBuildBuilder(MSBuildBuilder):
+    @property
+    def build_directory(self):
+        def get_file_string_number(f):
+            s = re.findall(r"\d+$", f)
+            return (int(s[0]) if s else -1, f)
+
+        win_dir = os.path.join(super().build_directory, "windows")
+        compiler_dirs = []
+        with working_dir(win_dir):
+            for obj in os.scandir():
+                if obj.is_dir():
+                    compiler_dirs.append(obj.name)
+        newest_compiler = max(compiler_dirs, key=get_file_string_number)
+        return os.path.join(win_dir, newest_compiler)
+
+    def is_64bit(self):
+        return platform.machine().endswith("64")
+
+    def msbuild_args(self):
+        plat = "x64" if self.is_64bit() else "x86"
+        if self.pkg.spec.satisfies("libs=shared,static"):
+            f = "xz_win.sln"
+        elif self.pkg.spec.satisfies("libs=shared"):
+            f = "liblzma_dll.vcxproj"
+        else:
+            f = "liblzma.vcxproj"
+        return [self.define("Configuration", "Release"), self.define("Platform", plat), f]
+
+    def install(self, pkg, spec, prefix):
+        with working_dir(self.build_directory):
+            # Ensure we have libs directory
+            if not os.path.isdir(prefix.lib):
+                mkdirp(prefix.lib)
+            libs_to_find = []
+            if self.pkg.spec.satisfies("libs=shared,static"):
+                libs_to_find.extend(["*.dll", "*.lib"])
+            elif self.pkg.spec.satisfies("libs=shared"):
+                libs_to_find.append("*.dll")
+            else:
+                libs_to_find.append("*.lib")
```

Comment:
My recollection is that `.dll` files are not used for linking, so we'll want to copy over all `.lib` files regardless of the value of the `libs` variant. I suppose if `libs=static`, we might not want to copy the `.dll` files. If they are built though, I'm inclined to copy them over. It might make sense to not provide the `libs` variant on Windows (since it looks like it doesn't affect what is built).

Reply:
If `libs=static` there should be no .dll files and if there are we shouldn't be copying them.  Good catch on the not copying libs. Will make those changes. The variant does impact what is built... where are you seeing it doesn't change what is built, perhaps I'm overlooking something.

Reply:
> The variant does impact what is built... where are you seeing it doesn't change what is built, perhaps I'm overlooking something.

Sorry yes I missed that, so the variant should stay.